### PR TITLE
Upgrade pre-commit hooks

### PR DESCRIPTION
**Pull request by asottile**

### CODE REVIEW

File: doc/en/example/assertion/failure_demo.py [hunk 1]
```diff
@@ -245,9 +245,9 @@ class A(object):
             a = 1
 
         b = 2
-        assert (
-            A.a == b
-        ), "A.a appears not to be b\n" "or does not appear to be b\none of those"
+        assert A.a == b, (
+            "A.a appears not to be b\n" "or does not appear to be b\none of those"
```
Comment (RonnyPfannschmidt) [inline doc/en/example/assertion/failure_demo.py]: please join the 2 strings

File: testing/python/approx.py [hunk 2]
```diff
@@ -62,11 +62,11 @@ def test_repr_string(self, plus_minus):
     @pytest.mark.parametrize(
         "value, repr_string",
         [
-            (5., "approx(5.0 {pm} 5.0e-06)"),
-            ([5.], "approx([5.0 {pm} 5.0e-06])"),
-            ([[5.]], "approx([[5.0 {pm} 5.0e-06]])"),
-            ([[5., 6.]], "approx([[5.0 {pm} 5.0e-06, 6.0 {pm} 6.0e-06]])"),
-            ([[5.], [6.]], "approx([[5.0 {pm} 5.0e-06], [6.0 {pm} 6.0e-06]])"),
+            (5.0, "approx(5.0 {pm} 5.0e-06)"),
```
Comment (RonnyPfannschmidt) [inline testing/python/approx.py]: those are really interesting

File: doc/en/example/assertion/failure_demo.py (hunk 1 above)
Comment (asottile) [reply doc/en/example/assertion/failure_demo.py]: going to leave this, we can fix these all later:
```
$ git grep '"[^"]*" "[^"]*"'
doc/en/example/assertion/failure_demo.py:        ), "A.a appears not to be b\n" "or does not appear to be b\none of those"
doc/en/example/reportingdemo.rst:            ), "A.a appears not to be b\n" "or does not appear to be b\none of those"
src/_pytest/assertion/util.py:                u("Skipping %s identical leading " "characters in diff, use -v to show")
src/_pytest/cacheprovider.py:            tw.line("%s contains unreadable content, " "will be ignored" % key)
src/_pytest/capture.py:        raise UnsupportedOperation("redirected stdin is pseudofile, " "has no fileno()")
src/_pytest/compat.py:            ("could not find real function of {start}" "\nstopped at {current}").format(
src/_pytest/deprecated.py:    "pycollector makeitem was removed " "as it is an accidentially leaked internal api"
src/_pytest/helpconfig.py:        "[pytest] ini-options in the first " "pytest.ini|tox.ini|setup.cfg file found:"
src/_pytest/mark/evaluate.py:                            msg = "you need to specify reason=STRING " "when using booleans as conditions."
src/_pytest/python.py:        help="prefixes or glob names for Python test function and " "method discovery",
src/_pytest/recwarn.py:        msg = "exceptions must be old-style classes or " "derived from Warning, not %s"
src/_pytest/terminal.py:        "failed passed skipped deselected " "xfailed xpassed warnings error"
testing/acceptance_test.py:                "def test_{}(): pass\n" "def test_other():pass".format(dirname)
testing/acceptance_test.py:            "def test_bar(): pass\n" "def test_other(a_fixture):pass"
testing/acceptance_test.py:            "import pytest\n" "@pytest.fixture\n" "def a_fixture():pass"
testing/logging/test_formatter.py:        "dummypath                   10 " "\x1b[32mINFO    \x1b[0m Test Message"
testing/logging/test_formatter.py:    assert output == ("dummypath                   10 " "INFO     Test Message")
testing/test_config.py:            args=123 "123 hello" "this"
testing/test_junitxml.py:            classname="xyz.test_junit_prefixing." "TestHello",
testing/test_pdb.py:            "--show-capture=all --pdb " "-p no:logging %s" % p1
```

**Comment by asottile** [comment]
hrmmm, seem to be hitting https://github.com/pre-commit/pre-commit/issues/835 (which itself is an intersection of pip / virtualenv / black / windows)

**Comment by asottile** [comment]
I *think* this might magically fix itself when `virtualenv` makes a release.  So I'm going to wait on that :)

File: doc/en/example/assertion/failure_demo.py (hunk 1 above)
Comment (RonnyPfannschmidt) [reply doc/en/example/assertion/failure_demo.py]: @asottile i agree,  please open a followup issue

**Comment by RonnyPfannschmidt** [comment]
great work :)

**Comment by asottile** [comment]
@RonnyPfannschmidt oh uh, this breaks the `linting` env on windows -- I wasn't quite ready to merge it yet 😆 

couple of options:
- temporarily disable windows (should be fixed by the end of the week once `virtualenv` releases if my hunch is right)
- upgrade the `linting` env to python3.7 (should use UTF-8 mode I think?)
- revert-revert and then I can re-apply later

File: doc/en/example/assertion/failure_demo.py (hunk 1 above)
Comment (asottile) [reply doc/en/example/assertion/failure_demo.py]: done! https://github.com/pytest-dev/pytest/issues/4093

**Comment by RonnyPfannschmidt** [comment]
@asottile oh, my bad - lets disable linting on windows and create a followup issue for after virtualenv releases

**Comment by asottile** [comment]
no problem! I made #4094 and #4095 to handle this

**Pull request merged**